flatfs.go

```diff
@@ -97,6 +97,7 @@ func combineAccuracy(a, b initAccuracy) initAccuracy {
 }
 
 var _ datastore.Datastore = (*Datastore)(nil)
+var _ datastore.PersistentDatastore = (*Datastore)(nil)
```

Comment:
We need Batching here too, right? Do we also need any type checks on the batch type?